### PR TITLE
Import foreman_probing_core

### PR DESCRIPTION
**Pull request by adamruzicka**

### CODE REVIEW

File: lib/foreman_probing_core.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+require 'dynflow'
+require 'foreman_tasks_core'
+
+module ForemanProbingCore
+  extend ForemanTasksCore::SettingsLoader
+
+  if ForemanTasksCore.dynflow_present?
```
Comment (ekohl) [inline lib/foreman_probing_core.rb]: Should this be updated now? AFAIK Dynflow should always be present and we want to get rid of `ForemanTasksCore`, right?

File: lib/smart-proxy-probing.rb [hunk 2]
```diff
@@ -2,9 +2,23 @@
...
+      def use_nmap?
+        # TODO:
+        # Settings.nmap_enabled &&
+        nmap_available?
+      end
+
+      def nmap_available?
+        return @nmap_available unless @nmap_available.nil?
+        `nmap`
+        @nmap_available = true
```
Comment (ekohl) [inline lib/smart-proxy-probing.rb:18]: Should this use `Proxy::Util::which('nmap')` instead?

File: smart-proxy-probing.gemspec [hunk 3]
```diff
@@ -21,5 +21,7 @@ Gem::Specification.new do |gem|
   gem.require_paths    = ["lib"]
   gem.license = 'GPLv3'
 
+  s.add_dependency('foreman-tasks-core', '~> 0.1')
+  s.add_dependency('ruby-nmap', '~> 0.9')
```
Comment (ekohl) [inline smart-proxy-probing.gemspec]:
```suggestion
  gem.add_dependency('foreman-tasks-core', '~> 0.1')
  gem.add_dependency('ruby-nmap', '~> 0.9')
```

File: lib/smart-proxy-probing/actions.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+require 'dynflow'
+
+module Proxy::Probing
+  module Actions
+    require 'foreman_probing_core/actions/use_probe'
```
Comment (ekohl) [inline lib/smart-proxy-probing/actions.rb]: This require doesn't look correct anymore.

File: lib/smart-proxy-probing/probes.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+module Proxy::Probing
+  module Probes
+    require 'foreman_probing_core/probes/abstract'
+    require 'foreman_probing_core/probes/nmap'
+    require 'foreman_probing_core/probes/tcp'
+    require 'foreman_probing_core/probes/udp'
+    require 'foreman_probing_core/probes/icmp'
```
Comment (ekohl) [inline lib/smart-proxy-probing/probes.rb]: These don't look correct.

**Review by ekohl — CHANGES_REQUESTED** [review]

**Review by ekohl — APPROVED** [review]

**Comment by adamruzicka** [comment]
Thank you @ekohl !

**Pull request merged**